qbraid_qir/qiskit/elements.py

```
@@ -0,0 +1,189 @@
+from qiskit.circuit import Clbit, Qubit
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+
+if TYPE_CHECKING:
+    from qiskit.circuit.instruction import Instruction
+
+
+class _QuantumCircuitElement(metaclass=ABCMeta):
+    """Abstract base class for quantum circuit elements."""
+
+    @classmethod
+    def from_element_list(cls, elements):
```

Ruff (0.15.2)

[warning] Missing return type annotation for classmethod `from_element_list`

(ANN206)